middlewares/ip_witelister.go

```diff
@@ -0,0 +1,85 @@
+package middlewares
+
+import (
+	"fmt"
+	"github.com/codegangsta/negroni"
+	"github.com/containous/traefik/log"
+	"net"
+	"net/http"
+)
+
+// IPWhitelister is a middleware that provides Checks of the Requesting IP against a set of Whitelists
+type IPWhitelister struct {
+	handler    negroni.Handler
+	whitelists []*net.IPNet
+}
+
+// NewIPWhitelister builds a new IPWhitelister given a list of CIDR-Strings to whitelist
+func NewIPWhitelister(whitelistStrings []string) (*IPWhitelister, error) {
+	whitelister := IPWhitelister{}
+
+	if len(whitelistStrings) == 0 {
+		return nil, fmt.Errorf("no whitelists provided")
+	}
+
+	for _, whitelistString := range whitelistStrings {
+		_, whitelist, err := net.ParseCIDR(whitelistString)
+		if err != nil {
+			return nil, fmt.Errorf("parsing CIDR whitelist %s: %+v", whitelist, err)
```

Comment:
Error messages should just be formatted using `%v` or `%s` (no plus sign) as they are meant for humans.